pyomo/core/tests/unit/test_componentuid.py

```diff
@@ -0,0 +1,891 @@
+
+    def test_find_wildcard_not_exists(self):
+        cuid = ComponentUID('b[*,*].c.x')
+        self.assertIsNone(cuid.find_component_on(self.m))
+
+    # def test_find_implicit_exists(self):
+    #     cuid = ComponentUID('b:1,2.c.a:3')
+    #     self.assertTrue(cuid.find_component_on(self.m) is
+    #                     self.m.b[1,'2'].c.a[3])
+
+    def test_find_implicit_notExists_1(self):
+        cuid = ComponentUID('b:1,2.c.a:4')
+        self.assertTrue(cuid.find_component_on(self.m) is None)
+
+    def test_find_implicit_notExists_2(self):
+        cuid = ComponentUID('b:1,1.c.a:3')
+        self.assertTrue(cuid.find_component_on(self.m) is None)
+
+    def test_find_explicit_notExists_1(self):
+        cuid = ComponentUID('b:1,2.c.a:$3')
+        self.assertTrue(cuid.find_component_on(self.m) is None)
+
+    def test_find_explicit_notExists_2(self):
+        cuid = ComponentUID('b:$1,2.c.a:3')
+        self.assertTrue(cuid.find_component_on(self.m) is None)
+
+    def test_printers_1(self):
+        cuid = ComponentUID(self.m.b[1,'2'].c.a[3])
+        s = "b[1,'2'].c.a[3]"
+        r1 = "b:#1,$2.c.a:#3"
+        r2 = "b[1,'2'].c.a[3]"
+        self.assertEqual(str(cuid), s)
+        self.assertEqual(repr(cuid), r2)
+        self.assertEqual(cuid.get_repr(1), r1)
+        self.assertEqual(cuid.get_repr(2), r2)
+        with self.assertRaisesRegex(
+                ValueError, "Invalid repr version '3'; expected 1 or 2"):
+            cuid.get_repr(3)
+
+    def test_printers_2(self):
+        cuid = ComponentUID('b:$1,2.c.a:#3')
+        s = "b['1',2].c.a[3]"
+        r1 = "b:$1,#2.c.a:#3"
+        r2 = "b['1',2].c.a[3]"
+        self.assertEqual(str(cuid), s)
+        self.assertEqual(repr(cuid), r2)
+        self.assertEqual(cuid.get_repr(1), r1)
+        self.assertEqual(cuid.get_repr(2), r2)
+
+    def test_printers_3(self):
+        cuid = ComponentUID('b:**.c.a:*')
+        s = 'b[**].c.a[*]'
+        r1 = "b:**.c.a:*"
+        r2 = "b[**].c.a[*]"
+        self.assertEqual(str(cuid), s)
+        self.assertEqual(repr(cuid), r2)
+        self.assertEqual(cuid.get_repr(1), r1)
+        self.assertEqual(cuid.get_repr(2), r2)
+
+    def test_printers_4(self):
+        cuid = ComponentUID('b:*,*.c.a:**')
+        s = 'b[*,*].c.a[**]'
+        r1 = "b:*,*.c.a:**"
+        r2 = "b[*,*].c.a[**]"
+        self.assertEqual(str(cuid), s)
+        self.assertEqual(repr(cuid), r2)
+        self.assertEqual(cuid.get_repr(1), r1)
+        self.assertEqual(cuid.get_repr(2), r2)
+
+    def test_matches_explicit(self):
+        cuid = ComponentUID(self.m.b[1,'2'].c.a[3])
+        self.assertTrue(cuid.matches(self.m.b[1,'2'].c.a[3]))
+        self.assertFalse(cuid.matches(self.m.b[1,'2'].c.a['2']))
+
+    # def test_matches_implicit(self):
+    #     cuid = ComponentUID('b:1,2.c.a:3')
+    #     self.assertTrue(cuid.matches(self.m.b[1,'2'].c.a[3]))
+    #     self.assertFalse(cuid.matches(self.m.b[1,'2'].c.a['2']))
```

Comment:
Why is this test commented out?

Reply:
See above comment: ComponentUID no longer supports "fuzzy" matching of ambiguous strings.